locker/src/main/java/org/killbill/commons/locker/GlobalLockBase.java

```diff
@@ -59,10 +59,14 @@ public Void execute() throws RuntimeException {
 
                 if (resetCallback != null && !resetCallback.reset(lockName)) {
                     // We are not the last one using that lock, bail early (AND don't close the connection)...
+                    logger.info("Skip releasing lock (owner re-entered) for " + lockName);
                     return null;
                 }
                 try {
-                    lockDao.releaseLock(connection, lockName);
+                    final boolean status = lockDao.releaseLock(connection, lockName);
+                    if (!status) {
+                        logger.warn("Failed to release global lock for " + lockName);
```

Comment:
If we indeed see this, we should update the DAO interface to return the actual value return by MySQL (`0 / null`): https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_release-lock